### ISSUE TITLE
Canceling ChangeBlockEvent.Break causes block to lose custom data

### ISSUE THREAD

**Issue by NickAcPT** [SpongePowered/Sponge#2359]
**I am currently running**

- SpongeVanilla version: 1.12.2-7.1.7-RC272
- Java version: Java(TM) SE Runtime Environment (build 1.8.0_181-b13)

- Plugins/Mods: Just my plugin

**Issue Description**
I have added custom data to a mobspawner block and it gets persisted properly.
But, when I setup an event listener of `ChangeBlockEvent.Break` that cancels the break of the block and have it cancel the event and mark all transactions with invalid, it restores the block but there's no data attached anymore.

With the help of the debugger, I was able to see that the data was there when I first attempted to break the block.
After attempting to break it a second time, it was no longer there and was just lost.

**Comment by JBYoshi** [comment]
Could you post the code you use to create the `DataRegistration` object? This happened for me at first, but it was just caused by a problem with my test plugin while creating the `DataRegistration`.

**Comment by NickAcPT** [comment]
```java
        DataRegistration.builder()
                .name("Custom Item")
                .id("custom_item")
                .dataClass(CustomItemData.class)
                .immutableClass(CustomItemData.Immutable.class)
                .builder(new CustomItemData.Builder())
                .build();
```

That is the code that registers the custom data. It persists fine even after a restart. It's just not rolled back from the block snapshot.

It appears to me that the problem is not on my code, but on the code that does the snapshot rollback.

**Comment by JBYoshi** [comment]
<s>You'll need to put that in an event handler for `GameRegistryEvent.Register<DataRegistration>` and pass that to `event.register()`. That's the same problem I had - it doesn't register when you call `build()`.</s>

**Comment by JBYoshi** [comment]
...Actually, never mind. `build()` actually does register. My problem was that I forgot to call `.id()` at first.

**Comment by JBYoshi** [comment]
Does anything show up in the logs when you break the block?

**Comment by NickAcPT** [comment]
Nope. No error or anything. It just doesn't rollback the custom data.

**Comment by JBYoshi** [comment]
Quick question - does this still happen on RC277?

I saw some of your code on GitHub and downloaded it. The latest version I could find works properly in my development environment. (I used `/customitems` to give myself a block inspector and a coal generator. After placing the coal generator, I disabled all block changes and attempted to break the block. Then I used the block inspector on the coal generator, and it still showed "coal generator" in the console.)
